test/unit/node_test.rb

```diff
@@ -58,6 +58,16 @@ class NodeTest < ActiveSupport::TestCase
       assert_equal nodes.length, Tag.find_nodes_by_type('awesome', ['note', 'page']).length
     end
   end
+  
+  test 'node mysql native fulltext search returning two tag-based matches' do
+    assert Node.count > 0
+    if ActiveRecord::Base.connection.adapter_name == 'Mysql2'
+      nodes = Node.search(query: 'awesome',query: 'cool', limit: 1000)
+      assert_not_nil nodes
+      assert nodes.length > 0
+      assert_equal nodes.length, Tag.find_nodes_by_type('awesome','cool', ['note', 'page']).length
```

Comment:
Likewise, i don't think `Tag.find_nodes_by_type` is set up to accept 2 separate strings. So this part of the test might need to be simpler -- just looking through the test data in `/test/fixtures/` to find a test node in the test fixtures that has 2 tags. Then the test would be expected to find just one node, and you could even do a comparison like:

```ruby
assert_equal 1, nodes.length
assert_equal nodes(:somenodefromfixtures), nodes.first
```

see how you can fetch a node from fixtures with the `nodes(:_____)` syntax. Fixtures are our test data and are described in flat `.yml` files and are loaded into the database when tests are run. Make sense? 

Great start here! Nicely written test! 👍

Reply:
I think I understand... I tried to do this in the changes :)